reporter/internal/pdata/pdata.go

```diff
@@ -0,0 +1,63 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package pdata // import "go.opentelemetry.io/ebpf-profiler/reporter/internal/pdata"
+
+import (
+	lru "github.com/elastic/go-freelru"
+
+	"go.opentelemetry.io/ebpf-profiler/libpf"
+	"go.opentelemetry.io/ebpf-profiler/libpf/xsync"
+	"go.opentelemetry.io/ebpf-profiler/reporter/internal/samples"
+)
+
+// Pdata holds the cache for the data used to generate the events reporters
+// will export when handling OTLP data.
+type Pdata struct {
+	// samplesPerSecond is the number of samples per second.
+	samplesPerSecond int
+
+	// Executables stores metadata for executables.
+	Executables *lru.SyncedLRU[libpf.FileID, samples.ExecInfo]
+
+	// Frames maps frame information to its source location.
+	Frames *lru.SyncedLRU[
+		libpf.FileID,
+		*xsync.RWMutex[map[libpf.AddressOrLineno]samples.SourceInfo],
+	]
+
+	// ExtraSampleAttrProd is an optional hook point for adding custom
+	// attributes to samples.
+	ExtraSampleAttrProd samples.SampleAttrProducer
+}
+
+func New(sps int, executablesCacheElements, framesCacheElements uint32,
```

Comment:
As all other arguments are also written in their extended form, `sps` should also be `samplesPerSecond`. WIthout implementation knowledges, `sps` could be anything otherwise.
```suggestion
func New(samplesPerSecond int, executablesCacheElements, framesCacheElements uint32,
```